pkgs/build-support/dotnet/add-nuget-deps/default.nix

```diff
@@ -12,24 +12,55 @@
 }:
 
 {
+  /**
+    A list of nuget packages.
+
+    Should be a list of derivations or arguments to be applied to `fetchNupkg`.
+
+    ```nix
+    [
+      # Arguments for use fetchNupkg
+      { pname = "FosterFramework"; version = "0.1.15-alpha"; hash = "sha256-lM6eYgOGjl1fx6WFD7rnRi/YAQieM0mx60h0p5dr+l8="; }
+
+      # Or a derivation
+      (fetchNupkg { pname = "Microsoft.AspNetCore.App.Runtime.linux-x64"; version = "8.0.1"; hash = "sha256-QbUQXjCzr8j8u/5X0af9jE++EugdoxMhT08F49MZX74="; })
+    ]
+    ```
+
+    Either a JSON or nix file with the same structure can also be used.
+  */
   nugetDeps,
   overrideFetchAttrs ? x: { },
 }:
 fnOrAttrs: finalAttrs:
 let
   attrs = if builtins.isFunction fnOrAttrs then fnOrAttrs finalAttrs else fnOrAttrs;
 
-  deps =
-    if (nugetDeps != null) then
-      if lib.isDerivation nugetDeps then
-        [ nugetDeps ]
-      else if lib.isList nugetDeps then
-        nugetDeps
-      else
-        assert (lib.isPath nugetDeps);
-        callPackage nugetDeps { fetchNuGet = fetchNupkg; }
+  callNupkg = pkg: if lib.isDerivation pkg then pkg else fetchNupkg pkg;
+
+  loadDeps =
+    x:
+    if x == null then
+      [ ]
+    else if lib.isDerivation x then
+      [ x ]
+    else if builtins.isList x then
+      builtins.map callNupkg x
```

Comment:
Yes. But that's intentional, because JSON cannot contain actual derivations.

This allows the list to contain _either_ derivations or arguments passed to `fetchNupkg`.

Reply:
Maybe a better name for `callNupkg` would be `coerceDepToDrv`, `coerceToPkg`, or similar?

Reply:
I've converted this back to the original format so we don't end up supporting more things than the original.

If people want to add derivation-based nuget packages as if they were packages then they can use `buildInputs`.